Add helm registry mods and redirects (#1046)
spectrocloud/librarium: content/docs/09-registries-and-packs/6-helm-charts.md

---
title: "Helm Registries"
metaTitle: "Helm Registries"
metaDescription: "Palette Dev Engine Helm Registries"
hideToC: false
fullWidth: false
---

import Tabs from 'shared/components/ui/Tabs';
import InfoBox from 'shared/components/InfoBox';
import WarningBox from 'shared/components/WarningBox';
import PointsOfInterest from 'shared/components/common/PointOfInterest';
import Tooltip from "shared/components/ui/Tooltip";

# Palette Helm Registry
Helm Charts are a collection of Kubernetes resource files capable of deploying services of varying complexity. Palette provides some stable default Helm charts in its public Helm Chart Registry. 

> Palette also leverages developers' custom Helm registries. You can add your own public or private Helm registries to Palette's Helm registry.

Reviewer: ```suggestion
Palette also supports creating custom Helm registries. You can add your own public or private Helm registries to Palette's Helm registry.
```
I think we can avoid specifying a specific person and keep it generic. There are other roles besides developers that use Palette. Many of them are system administrators or infrastructure analysts.

> The Helm Chart registry synchronizes charts with the Palette console so you can use them when you create cluster profiles.

Reviewer: ```suggestion
The Helm Chart registry synchronizes charts with Palette, so you can use them when you create cluster profiles.
```
Let's leave out the console. The console is really the UI and not the driver of logic. So we can just say Palette 😄

> ## Prerequisite

Reviewer: ```suggestion
# Prerequisite
```
This should always be a major heading. That way, users can find it in the right-hand-side navigation menu.

For security, Palette requires Helm OCI version 3.7.0 and higher.

> ## Add a Helm Chart Registry to Palette

Reviewer: ```suggestion
# Add a Helm Chart Registry to Palette
```
Per how-to style guide.

To add your private Helm Chart Registry to Palette:

> 1. Log in as a Tenant administrator.

Reviewer: ```suggestion
1. Log in to Palette as a Tenant administrator. 
```

2. From the **Main Menu** navigate to **Tenant Settings > Registries**.

3. From the **Helm Registries** tab, click **Add New Helm Registry** and type the registry name and endpoint. If the registries list is long, you may need to scroll down to see the Add link.

3. Type the name of your registry and its endpoint.

> 3. Choose ``Protected`` mode based on whether your network is private or public:

Reviewer: @ritawatson should this be in bold instead of code blocks? It seems like a UI element but I could be wrong.

   * Toggle ``Protected`` mode to *on* if your Helm registry is deployed in a private network. Palette downloads and deploys charts from protected chart registries instead of scanning a private network for them.

      When your registry is protected:

>       * Charts are not synchronized with the Palette console, and you must type Helm chart names and versions when you create Cluster Profiles.

Reviewer: ```suggestion
      * Charts are not synchronized with Palette, and you must type Helm chart names and versions when you create Cluster Profiles.
```

      * The **Last Synced** column in the **Helm Registries** tab displays *n/a*. 
   <br />
   <br />

    * Leave ``Protected`` mode toggled *off* if your Helm registry is deployed in a public network. We refer to Helm registries with this option disabled as being unprotected.  
    
      When your registry is unprotected: 

      * Palette synchronizes Helm charts with the console so you can select charts and versions from drop-down menus. 
      * The **Last Synced** column in the **Helm Registries** tab displays the date that charts were last synchronized in the Palette console.
   <br />

5. If you haven’t set up credentials for your registry, leave **No Authentication** toggled *on*.

   If your registry has credentials, toggle ``No Authentication`` to *off* and type the registry **Username** and **Password**.
<br />

7. Confirm your changes. 

Your chart is now deployed in Palette's Helm Chart Registry and is ready to model in cluster profiles.

> ## Validation

Reviewer: ```suggestion
# Validation
```
Major heading, per the how-to style guide
